app/editor/src/features/admin/topics/TopicList.tsx

```diff
@@ -59,40 +55,89 @@ export const TopicList: React.FC = () => {
+    changeParentOverflow();
+    window.addEventListener('resize', updateMaxHeight);
+  }, []);
+
+  React.useEffect(() => {
+    if (items.length && !loading) updateMaxHeight();
+  }, [items.length, loading]);
+
   React.useEffect(() => {
     const topic = items.find((i) => i.id === topicId) ?? defaultTopic;
     setTopic(topic);
   }, [topicId, items]);
 
+  const handleRemove = async (topicId: number) => {
+    const topic = items.find((i) => i.id === topicId);
+    if (!topic) return;
+    const removedTopic = { ...topic, isEnabled: false };
+    await api.updateTopic(removedTopic);
+    setItems(items.filter((t) => t.id !== topic.id));
+    setTopics(topics.map((item) => (item.id === topic.id ? removedTopic : item)));
+    if (!!id) navigate('/admin/topics');
+    toast.success(`${topic.name} has successfully been deleted.`);
+  };
+
   const handleSubmit = async (values: ITopicModel) => {
     try {
       var results = [...items];
+      let updatedTopics: ITopicModel[];
+      const existsTopic = topics.find((x) => x.name === values.name);
       if (values.id === 0) {
-        const topic = topics.find((x) => x.name === values.name);
-        if (!topic) {
+        if (!existsTopic) {
           const result = await api.addTopic(values);
           results = [...items, result];
+          updatedTopics = [...topics, result];
         } else {
           const result = await api.updateTopic({
-            ...topic,
+            ...existsTopic,
             isEnabled: values.isEnabled,
             description: values.description,
             topicType: values.topicType,
           });
-          if (topic.isEnabled) results = items.map((i) => (i.id === result.id ? result : i));
+          if (existsTopic.isEnabled) results = items.map((i) => (i.id === result.id ? result : i));
           else if (values.isEnabled) results = [...items, result];
+          updatedTopics = topics.map((i) => (i.id === existsTopic.id ? result : i));
         }
       } else {
-        const result = await api.updateTopic(values);
-        results = items.map((i) => (i.id === values.id ? result : i));
+        if (!existsTopic || values.id === existsTopic.id) {
+          const result = await api.updateTopic(values);
+          results = items.map((i) => (i.id === values.id ? result : i));
+          updatedTopics = topics.map((i) => (i.id === topic.id ? result : i));
+        } else {
+          toast.warn(`${values.name} already exists.`);
+          return;
+        }
       }
       setItems(results.filter((x) => x.isEnabled));
+      setTopics(updatedTopics);
       toast.success(`${values.name} has successfully been saved.`);
+      // Do we have a better way here for the clean-up after adding a new topic?
+      if (!id || id === '0') {
+        values.name = defaultTopic.name;
+        values.topicType = defaultTopic.topicType;
+      }
     } catch {
       // Ignore error as it's handled globally.
     }
   };
 
+  const updateMaxHeight = () => {
```

Comment:
Can you explain what the purpose of this function is?  It looks like you're increasing the size of the table, but I don't know why we would ever want to do this.

Reply:
I think this function will go away after @CelloLiu makes changes from feedback that came out of the demo.  There is no need to window box the table with a scroll bar and the "new topic" section will be above it now.

Reply:
Removed based on the demo feedbacks, thanks